src/main/java/com/vmware/vipclient/i18n/messages/api/opt/local/LocalPatternOpt.java

```diff
@@ -22,33 +22,38 @@
 public class LocalPatternOpt implements PatternOpt{
     Logger logger = LoggerFactory.getLogger(LocalPatternOpt.class);
 
-    public JSONObject getPatterns(String locale) {
+    public void getPatterns(String locale, PatternCacheItem cacheItem) {
         logger.debug("Look for pattern from local bundle for locale [{}]!", locale);
         String normalizedLocale = CommonUtil.getCLDRLocale(locale, localePathMap, localeAliasesMap);
         logger.debug("Normalized locale for locale [{}] is [{}]", locale, normalizedLocale);
-        return getPatternsByLocale(normalizedLocale);
+        getPatternsByLocale(normalizedLocale, cacheItem);
     }
 
     @Override
-    public JSONObject getPatterns(String language, String region) {
+    public void getPatterns(String language, String region, PatternCacheItem cacheItem) {
         logger.debug("Look for pattern from local bundle for language [{}], region [{}]!", language, region);
         LocaleDataDTO resultData = CommonUtil.getLocale(language, region);
         String normalizedLocale = resultData.getLocale();
         logger.debug("Normalized locale for language [{}], region [{}] is [{}]", language, region, normalizedLocale);
-        return getPatternsByLocale(normalizedLocale);
+        getPatternsByLocale(normalizedLocale, cacheItem);
     }
 
-    private JSONObject getPatternsByLocale(String normalizedLocale) {
+    private void getPatternsByLocale(String normalizedLocale, PatternCacheItem cacheItem) {
         if(normalizedLocale == null || normalizedLocale.isEmpty())
-            return null;
+            return;
         try {
             String patternStr = PatternUtil.getPatternFromLib(normalizedLocale, null);
             Map<String, Object> patterns = (Map<String, Object>) new JSONParser().parse(patternStr);
-            return (JSONObject) patterns.get(PatternKeys.CATEGORIES);
+            if(patterns != null) {
+                logger.debug("Find the pattern from local bundle for locale [{}].\n", normalizedLocale);
+                cacheItem.set((Map<String, Object>) patterns.get(PatternKeys.CATEGORIES), System.currentTimeMillis());
+            }else{
+                logger.debug("Doesn't find the pattern from local bundle for locale [{}].\n", normalizedLocale);
+                cacheItem.set(System.currentTimeMillis());
```

Comment:
```suggestion
           
```
Remove this line. We do not want to update anything in the cache item if fetch failed.

Reply:
This is for handling non-supported locales, just want to add an empty fetch result to cache, it's the same as you new an empty MessageCacheItem and add it to cache.